### PR TITLE
move display block strategy outsite the bundle

### PR DESCRIPTION
**Pull request by nicolasThal**

### CODE REVIEW

**Comment by itkg-ngilain** [comment]
I'm not sure about this one : this code is highly orchestra-dependant (several orchestra uses), and so can't be used in another project. So where is the interest to extract this from the bundle ?

**Comment by nicolasThal** [comment]
I think that there are multiple reason : 
 - By extracting the logic out of the bundle, the code is still dependant to open-orchestra but not to symfony
 - The media bundle (as created here) is linked to mongodb, with this separtion, we could create a MediaBundle linked to mysql to use all the logic we have created

File: MediaBundle/Resources/config/displayblock.yml [hunk 1]
```diff
@@ -16,11 +17,13 @@ services:
         arguments:
             - @request_stack
             - @open_orchestra_base.manager.tag
+            - 'OpenOrchestraMediaBundle:Block/Gallery:show.html.twig'
```
Comment (itkg-ngilain) [inline MediaBundle/Resources/config/displayblock.yml]: This should be a parameter to allows developers to override it

Comment (nicolasThal) [inline MediaBundle/Resources/config/displayblock.yml]: yes, good idea, i will do it :)

**Comment by itkg-ngilain** [comment]
I don't agree with you : the code rely both on orchestra and on symfony (use of httpfoundation). If we want to use alternative code, we can still play with the DI to set the good services.
I am not still convinced

**Comment by nicolasThal** [comment]
Yes, the code will still rely on some part of the symfony components (as you said HttpFoundation), but relying on some components is far from relying on the whole framework (cf `a year with symfoy`).

To use the media bundle with mysql instead of mongo, this allows you to load only the class you need and no class linked to mongo and doctrine odm

**Comment by itkg-ngilain** [comment]
I'm sorry, I'd like to understand your position but I don't see the link with the database/odm. Wherever the code is located, you still have to inject the @nodeRepository to the DisplayMediaStrategy or the @open_orchestra_media.repository.media to the MediaListByKeywordStrategy. What does it change ? Moreover, these repo could be using MySql, MongoDB or even Redis, they also are services 'overridable' ^^
I don't see the link between the database and the folder where the code is stored :/

**Pull request merged**